### PR TITLE
ci(workflows): Update versions

### PR DESCRIPTION
**Pull request by LuigiImVector**
## What does this PR do?
Improve repo

### Description

- Updated `actions/setup-node@v1` to [`v2`](https://github.com/actions/setup-node)
- Updated `actions/checkout@v1` to [`v2`](https://github.com/actions/checkout)
- Updated `ruby-version: 2.4.1` to [`2.6`](https://github.com/ruby/setup-ruby)

## Checklist:
- [X] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -4,14 +4,14 @@ jobs:
...
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v2
       - uses: trilom/file-changes-action@v1.2.4
         id: file_changes
         with:
           output: ''
       - uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.4.1
+          ruby-version: 2.6
```
Comment (LuigiImVector) [inline .github/workflows/main.yml:14]: In the official repository the [supported version of Ruby](https://github.com/ruby/setup-ruby/blob/51c0d5922252eab2cb7349f2142160df023fb656/README.md?plain=1#L17) are until `3.1.0` but in the [usage example](https://github.com/ruby/setup-ruby/blob/51c0d5922252eab2cb7349f2142160df023fb656/README.md?plain=1#L70) there is still `2.6`
Is it better to update the version to ` 3.1.0`?

**Comment by davorpa** [comment]
Note: this should be automated if we activate dependabot. See my PR #5521 :wink:

File: .github/workflows/main.yml (hunk 1 above)
Comment (davorpa) [reply .github/workflows/main.yml:14]: Ruby is used to check urls. Need check compatibility with that program

Comment (LuigiImVector) [reply .github/workflows/main.yml:14]: In the [awesome_bot example](https://github.com/dkhamsing/open-source-ios-apps/blob/master/.github/workflows/ruby.yml) it uses `ruby-version: 2.6.x`, I don't know if newest version are compatible

Comment (davorpa) [reply .github/workflows/main.yml:14]: > In the [awesome_bot example](https://github.com/dkhamsing/open-source-ios-apps/blob/master/.github/workflows/ruby.yml?rgh-link-date=2022-01-26T13%3A24%3A32Z) it uses `ruby-version: 2.6.x`, I don't know if newest version are compatible

Reviewing a bit more in documentation (https://github.com/dkhamsing/awesome_bot/blob/4989c0f248be71089ad300d93dd0b2abb930337e/.github/workflows/ci.yml#L15) test matrix is done over Ruby 2.6, 2.7 and 3.0

I completelly unknown Ruby but I think it's safe upgrade its version.

Comment (LuigiImVector) [reply .github/workflows/main.yml:14]: Maybe @dkhamsing can help us, what versions of ruby have compatibility with awesome_bot?

Comment (dkhamsing) [reply .github/workflows/main.yml:14]: awesome_bot ruby requirement is version 2, sorry if it's not better documented

Comment (LuigiImVector) [reply .github/workflows/main.yml:14]: Thanks for the clarification

Comment (davorpa) [reply .github/workflows/main.yml:14]: @LuigiImVector then alter some handled `.md` file to see how CI works and check after the differences

Comment (LuigiImVector) [reply .github/workflows/main.yml:14]: Seems that everything works correctly: [#2](https://github.com/LuigiImVector/free-programming-books/pull/2) (sorry for the confusing PR).

Comment (davorpa) [reply .github/workflows/main.yml:14]: Yep. It works. I imagine the amount of time elapsed to process all files :laughing: 

Workflow log on that PR: [logs_33.zip](https://github.com/EbookFoundation/free-programming-books/files/8063183/logs_33.zip)

**Review by davorpa — APPROVED** [review]
+1 for ruby version

**Pull request merged**